src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -204,7 +204,11 @@ private static async Task<ImmutableArray<DiagnosticData>> GetProjectStateDiagnos
                 if (documentId != null)
                 {
                     // file doesn't exist in current solution
-                    var document = project.Solution.GetDocument(documentId);
+                    var document = await project.Solution.GetDocumentAsync(
+                        documentId,
+                        includeSourceGenerated: project.Solution.Workspace.Services.GetService<ISyntaxTreeConfigurationService>() is { EnableOpeningSourceGeneratedFilesInWorkspace: true },
+                        cancellationToken).ConfigureAwait(false);
```

Comment:
Minor: This code pattern seems to be used at multiple places and is quite verbose. Probably deserves an extension method.

Reply:
It'll go away once we get rid of that feature flag that's controlling all of this. Not sure whether we'll get all the bugs fixed for 17.2, but hopefully by 17.3.